common/src/main/java/io/homeassistant/companion/android/common/data/websocket/WebSocketRequest.kt

```diff
@@ -24,4 +25,6 @@ data class WebSocketRequest(
     val eventTimeout: Long = 0L,
     val onEvent: Channel<Any>? = null,
     var onResponse: CancellableContinuation<SocketResponse>? = null
-)
+) {
+    val hasContinuationBeenInvoked = AtomicBoolean(onResponse == null)
```

Comment:
Setting this here seems to break the websocket because `onResponse` is null by default (as mentioned in the doc above the class definition), and as a result most of the native app functionality. It is only set [when the message is sent](https://github.com/home-assistant/android/blob/ece89d45c7177386fac34f6403b4d151c89a4cd5/common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt#L652).

(Everything will time out waiting for a response which never arrives.)

Reply:
I've set the initial value to false, which I believe resolves the issue. 

To test this I checked the list of available entities loads when adding a widget and that already added widgets load correctly. 

The property could be made immutable and the copy constructor used add the continuation. But since that would mean 2 objects are created for every message that's probably not the way to go.

Reply:
> I've set the initial value to false, which I believe resolves the issue.

Yes, looks like it. Feel free to mark the PR as ready to review so we can approve + merge.

> The property could be made immutable and the copy constructor used add the continuation. But since that would mean 2 objects are created for every message that's probably not the way to go.

That seems a bit wasteful and complicated when setting it to `false` by default also works.